### PR TITLE
Remove the ClusterRole aggregation from the default manifests

### PR DESCRIPTION
**Pull request by AlanGreene**
# Changes

Closes https://github.com/tektoncd/dashboard/issues/2047

ClusterRole aggregation is only used to support Dashboard extensions
and requires additional permissions on the cluster. This causes
problems in many scenarios and currently affects all users even if
they're not using extensions.

Move the aggregation rule behind an installer flag so users who
need it can still easily generate the required config but other
users benefit from the reduced permissions required for a default
install.

Document the flag in the extensions doc and describe the requirements
for achieving the same result manually for users not using the
installer script.

Also update the installer docs and help command to describe the
new option.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/dashboard/blob/main/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: docs/dev/installer.md [hunk 1]
```diff
@@ -63,24 +63,25 @@ Accepted commands:
 
 Accepted options:
         [--debug]                               Prints additional messages in the console
+        [--extensions-rbac]                     Enable ClusterRole aggregation for easier management of extensions RBAC
+        [--external-logs <logs-provider-url>]   External url to fetch logs from when not available in the cluster
```
Comment (briangleeson) [inline docs/dev/installer.md:67]: I feel like this sentence doesnt read right: `External url to fetch logs from when not available in the cluster`

Better? `External url from which to fetch logs, when logs are not available in the cluster`

File: docs/extensions.md [hunk 2]
```diff
@@ -94,6 +94,8 @@ rules:
 EOF
 ```
 
+Alternatively, if the `--extensions-rbac` is not used, ommit the label from the ClusterRole, and define the appropriate RoleBinding or ClusterRoleBinding yourself with the subject set to the `tekton-dashboard` service account.
```
Comment (briangleeson) [inline docs/extensions.md:97]: Typo for `omit`

File: scripts/installer [hunk 3]
```diff
@@ -476,24 +498,25 @@ help () {
   echo -e ""
   echo -e "Accepted options:"
   echo -e "\t[--debug]\t\t\t\tPrints additional messages in the console"
+  echo -e "\t[--extensions-rbac]\t\t\tEnable ClusterRole aggregation for easier management of extensions RBAC"
+  echo -e "\t[--external-logs <logs-provider-url>]\tExternal url to fetch logs from when not available in the cluster"
```
Comment (briangleeson) [inline scripts/installer:502]: Same comment as before with wording of this sentence

**Review by briangleeson — CHANGES_REQUESTED** [review]
Couple of comments on Strings

File: docs/extensions.md (hunk 2 above)
Comment (AlanGreene) [reply docs/extensions.md:97]: 🙈

**Pull request merged**